python/lsst/ts/ess/csc/accumulator/air_flow_accumulator.py

```diff
@@ -21,9 +21,20 @@
 
 __all__ = ["AirFlowAccumulator"]
 
+import math
+
 import numpy as np
 
-from .utils import get_median_and_std_dev
+from .utils import get_circular_mean_and_std_dev, get_median_and_std_dev
+
+
+# TODO DM-38119: delete this function and stop using it once the data type of
+# direction and directionStdDev is float.
+def float_as_int(value: float, nan_value: int = -1) -> int:
+    """Cast a float value to int, treating NaN specially."""
+    if math.isnan(value):
```

Comment:
It is a pity that a whole new import is needed when `np.isnan` would have worked equally well. Also note that returning a `NaN` is NOT compatible with `int`. Please return the value `-1` instead.

Reply:
This code does return -1 for nan. That is essentially the whole reason for this function, since otherwise one would use `int(round(value))`.

The advantage of math.isnan over np.isnan is that the math version does not have to check if the argument is an array. It's minor, but I strongly prefer to use math for known scalars and numpy for arrays. If nothing else, I think it gives the reader a useful clue. (That said, it is certainly possible to write some of these functions to handle arrays or scalars using numpy; I didn't bother to try with this function).

Reply:
Thanks, I completely failed to see that `nan_value` is a function argument. In all other code the value of `-1` is hard coded.